sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/Components/RecastNavigationComponent.cs

```diff
@@ -1,42 +1,56 @@
-﻿using DotRecast.Core.Numerics;
-using DotRecast.Detour;
-using Stride.BepuPhysics.Navigation.Definitions;
-using Stride.BepuPhysics.Navigation.Extensions;
-using Stride.BepuPhysics.Navigation.Processors;
+﻿using Stride.BepuPhysics.Navigation.Processors;
 using Stride.Core;
 using Stride.Core.Mathematics;
 using Stride.Engine;
 using Stride.Engine.Design;
 
 namespace Stride.BepuPhysics.Navigation.Components;
+/// <summary>
+/// This component is responsible for handling navigation-related properties and behaviors for an entity.
+/// </summary>
 [DataContract(nameof(RecastNavigationComponent))]
 [ComponentCategory("Bepu - Navigation")]
 [DefaultEntityComponentProcessor(typeof(RecastNavigationProcessor), ExecutionMode = ExecutionMode.Runtime)]
 public class RecastNavigationComponent : EntityComponent
 {
-    public float Speed { get; set; } = 5.0f;
+	/// <summary>
+	/// The speed at which the agent moves.
+	/// </summary>
+	public float Speed { get; set; } = 5.0f;
 
-    /// <summary>
-    /// True if a new path needs to be calculated, can be manually changed to force a new path to be calculated.
-    /// </summary>
-    [DataMemberIgnore]
-    public bool ShouldMove { get; set; } = true;
+	/// <summary>
+	/// Indicates if a new path needs to be calculated. Can be manually changed to force a new path calculation.
+	/// </summary>
+	[DataMemberIgnore]
+	public bool ShouldMove { get; set; } = true;
 
-    [DataMemberIgnore]
-    public bool SetNewPath { get; set; } = true;
+	/// <summary>
+	/// Indicates if a new path should be set.
+	/// </summary>
+	[DataMemberIgnore]
+	public bool SetNewPath { get; set; } = true;
 
-    [DataMemberIgnore]
-    public bool InSetPathQueue { get; set; }
+	/// <summary>
+	/// Indicates if the component is in the queue to set a new path.
+	/// </summary>
+	[DataMemberIgnore]
+	public bool InSetPathQueue { get; set; }
```

Comment:
Those three variables are a bit confusing, what is the difference between :
> Indicates if a new path needs to be calculated

And

> Indicates if a new path should be set.

I understand that `InSetPathQueue` occurs after one of those two, not sure which though.

Maybe this could be a get-only enum providing information on the progression of the component along its path, something like `{ Idle, Queued, Moving }`, and then users would have a function or property to set the destionation. Something like `NewDestination(Vector3)` or `Destination { get; set; }` where setting that property would force the entity's path to be discarded and set back in the queue ?

Reply:
That is a great point, for now I just fixed the comments but I will clean this up as I wasn't super happy with how I did it.